File: evals/lang/kotlin/project/shop/Checkout.kt
package shop

import shop.Product as Item
import kotlin.math.max

object Checkout {
    fun total(items: List<Item>, discountPct: Int): Int {
        val sum = items.sumOf { it.cents }
        return max(0, sum - (discountPct percentOf sum))
    }

    fun receipt(items: List<Item>): String =
        items.joinToString { it.label() } + " = " + total(items, 10)
}

File: evals/lang/python/project/shop/checkout.py
import shop.cart as cartmod
from shop.cart import Cart, Item
from shop.pricing import fetch_rates


def checkout(cart: Cart) -> float:
    rates = fetch_rates("eu")
    cart.add(Item("fee", 1.0))
    return cart.total()


def reset(cart: cartmod.Cart) -> None:
    cart.clear()
